geo/src/algorithm/bounding_rect.rs

```diff
@@ -1,13 +1,13 @@
 use crate::utils::{partial_max, partial_min};
 use crate::{
-    coord, CoordNum, Geometry, GeometryCollection, GeometryCow, Line, LineString, MultiLineString,
-    MultiPoint, MultiPolygon, Point, Polygon, Rect, Triangle,
+    coord, CoordNum, Coordinate, Geometry, GeometryCollection, GeometryCow, Line, LineString,
+    MultiLineString, MultiPoint, MultiPolygon, Point, Polygon, Rect, Triangle,
 };
 use geo_types::private_utils::{get_bounding_rect, line_string_bounding_rect};
 
 /// Calculation of the bounding rectangle of a geometry.
 pub trait BoundingRect<T: CoordNum> {
-    type Output;
+    type Output: Into<Option<Rect<T>>>;
```

Comment:
This was needed for some of the blanket Intersection implementations which now rely on BoundingRect.

Some geometries bbox is Rect, while others, like an empty Polygon, are Option<Rect>::None.